File: Mat/initializing.py
# -----------------------------------------------------------------------------
# 
#  Example for cv2.Mat initializing in Python3
# 
# -----------------------------------------------------------------------------

import cv2
import numpy as np

A = np.zeros((2, 2, 3), dtype=np.uint8)
A[:] = (0, 0, 255)

B = np.zeros((3, 3, 3), dtype=np.uint8)  # Cannot assign matrix values in this case
C = np.eye(4, 4, dtype=np.float64)  # MatLab style
D = np.zeros((2, 2), dtype=np.float32)  # MatLab style
E = np.ones((2, 2), dtype=np.uint8)  # MatLab style

F = np.array([[0, -1, 0], [-1, 5, -1], [0, -1, 0]], dtype=np.int32)  # For small matrices we have troubles...
G = np.array([0, -1, 0, -1, 5, -1, 0, -1, 0], dtype=np.float64).reshape(3, 3)  # For small matrices we have troubles...

R = np.zeros((400, 400, 3), dtype=np.uint8)  # Init Mat with random values
cv2.randu(R, 0, 255)  # Generate random values in R

print("\nA (common)\n", A)
print("\nB (NumPy)\n", B)
print("\nC (MatLab)\n", C)
print("\nD (CSV)\n", D)
print("\nE (C)\n", E)
print("\nF (Python)\n", F)
print("\nG (Default)\n", G)

cv2.imshow("R", R)  # Display image
cv2.waitKey()  # Wait for a key press
cv2.destroyAllWindows()  # Close all windows

File: Mat-special/initializing/initializing.cpp
//-----------------------------------------------------------------------------
//
// Example for cv::Mat initializing in C++ with OpenCV.
//
//-----------------------------------------------------------------------------

#include <opencv2/opencv.hpp>

#include <iostream>

int main() {
    cv::Mat A(2, 2, CV_8UC3, cv::Scalar(0, 0, 255));
    cv::Mat B, C, D, E, F, G, R;

    /* Cannot assign matrix values in this case. */
    B.create(3, 3, CV_8UC3);

    /* MatLab style. */
    C = cv::Mat::eye(cv::Size(4, 4), CV_64F);
    D = cv::Mat::zeros(cv::Size(2, 2), CV_32F);
    E = cv::Mat::ones(cv::Size(2, 2), CV_8UC1);

    /* For small matricies we have troubles... */
    F = (cv::Mat_<int>(3, 3) << 0, -1, 0, -1, 5, -1, 0, -1, 0);
    G = (cv::Mat_<double>({0, -1, 0, -1, 5, -1, 0, -1, 0})).reshape(3);

    /* Init Mat with random values. */
    R = cv::Mat::zeros(cv::Size(400, 400), CV_8UC3);
    cv::randu(R, cv::Scalar::all(0), cv::Scalar::all(255));

    std::cout << "\nA (common)\n" << A << '\n';
    std::cout << "\nB (NumPy)\n" << cv::format(B, cv::Formatter::FMT_NUMPY) << '\n';
    std::cout << "\nC (MatLab)\n" << cv::format(C, cv::Formatter::FMT_C) << '\n';
    std::cout << "\nD (CSV)\n" << cv::format(D, cv::Formatter::FMT_CSV) << '\n';
    std::cout << "\nE (C)\n" << cv::format(E, cv::Formatter::FMT_CSV) << '\n';
    std::cout << "\nF (Python)\n" << cv::format(F, cv::Formatter::FMT_PYTHON) << '\n';
    std::cout << "\nG (Default)\n" << cv::format(G, cv::Formatter::FMT_DEFAULT) << '\n';

    cv::imshow("R", R);

    cv::waitKey();
    cv::destroyAllWindows();
}
